### ISSUE TITLE
希望examples/qwen_image/model_training/lora/Qwen-Image-Blockwise-ControlNet-Inpaint.sh能加入对于多batch的支持

### ISSUE THREAD

**Issue by XIONGPEILIN** [modelscope/DiffSynth-Studio#982]
现在整个生成的pipeline 对于多个图片prompt输入都没有支持，我不是很熟悉如何给qwen2.5vl进行多batch的输入

**Comment by Artiprocher** [comment]
@XIONGPEILIN 

我们目前不打算提供 batch size > 1 的支持，原因如下：

- **更大的 batch size 无法实现显著加速**，由于 flash attention 等加速技术已经充分提高了 GPU 的利用率，因此更大的 batch size 只会带来更大的显存占用，无法带来显著加速

- **batch size > 1 需要全链路支持**，这是我们团队的人力无法实现的，单独一个 Qwen-Image 的 LoRA 训练支持 batch size > 1 是很容易的，但如果加上 ControlNet，各种 Adapter，各种输入哪怕有一个不支持 batch size > 1，整个训练链路就不支持 batch size > 1
  - 比如，Qwen-Image 的文本 embedding 是与文本长度相关的，虽然可以通过 padding 和 mask 对齐不同长度的 embedding，但会带来很大的维护成本
  - 比如，Qwen-Image 支持图像编辑之后，输入图的分辨率可以是不同的，那么就没法拼接成一个张量

- **未来的模型参数量会逐渐增大到单一 GPU 训练 batch size=1 训练都吃力**，工程上花更多精力支持 batch size > 1 的收益很快就会被磨掉，我们可以看到 FLUX 到 Qwen-Image，参数量从12B增加到了20B，模型参数量的增大是未来模型发展的必然趋势，所以
  - 我们希望把更多的精力放到多卡并行上，包括 Tensor parallel 和 sequential parallel，以及更适合预训练的 megatron 等并行技术

- **batch size > 1 目前存在很多等价实现**，包括
  - 多 GPU，多机器训练，global batch size = GPU 数量
  - Gradient Accumulation
    - Gradient Accumulation 在数学意义上和大 batch size 确实是等价的，大家可以自己写一个简单的神经网络框架试试，类似于 Pytorch，唯一需要注意的是 learning rate 需要除以 Gradient Accumulation 步数（如果是 SGD），或者在 loss 上除以 Gradient Accumulation 步数
    - 目前 Gradient Accumulation 主要面临的问题是计算精度，多步梯度累加起来会让梯度数值的范围变大，bf16 精度问题尤为明显，目前的训练框架 accelerator 没有在梯度累积时 scale bf16 精度的机制，会导致一定的精度问题（float32 则会好很多），但这是可以通过基础训练框架的进一步底层优化改善的

**Comment by XIONGPEILIN** [comment]
🙏回答
